shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -9,26 +9,39 @@ interface StepCompletionFeature {
         fun createState(step: Step, stepRoute: StepRoute): State =
             State(
                 currentStep = step,
-                continueButtonAction = if (stepRoute is StepRoute.Learn) {
-                    ContinueButtonAction.FetchNextStepQuiz
+                startPracticingAction = when (stepRoute) {
+                    is StepRoute.Learn.TheoryOpenedFromPractice, is StepRoute.Repeat.Theory ->
+                        StartPracticingAction.NavigateToBack
+                    is StepRoute.Learn.Step, is StepRoute.LearnDaily,
+                    is StepRoute.Repeat.Practice, is StepRoute.StageImplement ->
+                        StartPracticingAction.FetchNextStepQuiz
+                },
+                continuePracticingAction = if (stepRoute is StepRoute.Learn) {
+                    ContinuePracticingAction.FetchNextStepQuiz
                 } else {
-                    ContinueButtonAction.NavigateToBack
+                    ContinuePracticingAction.NavigateToBack
                 }
             )
     }
 
     data class State(
```

Comment:
Let's add currentStepRoute to the state. At the moment it is passed as an argument to the reducer for some reason despite it is passed to the `createState` method.